### PR TITLE
I did some stuff

### PR DESCRIPTION
**Pull request by squeezeboxyt**
# Pull request

 - [x] I have read and agreed to the [CODE OF CONDUCT](https://github.com/CascadeBot/CascadeBot/blob/master/CODE_OF_CONDUCT.md).
 - [x] What I'm implementing was assigned to me or was not being worked on and is a somewhat requested feature. For reference see our [trello](https://trello.com/b/gUQehy4l/bot).
 - [x] I have tested all of my changes.
 - [x] I have followed the style of the rest of the project.
 
## Added/Changed
 - I have proposed a ;shutdown command that can only be used by Developers, to shutdown the bot safely if needs be.
 - I have (hopefully) added some new fields to UserInfoCommand.java

## Feature
 - I think that the ;shutdown command should be implemented because it would be helpful in situations where the bots needs to be shutdown and nobody is around to access Console :man_shrugging:

### CODE REVIEW

**Comment by binaryoverload** [comment]
Take a moment to thing about what different fields we might want on this command. I'd suggest looking through the [User](https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/core/entities/User.html) and [Member](https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/core/entities/Member.html) objects

File: src/main/java/com/cascadebot/cascadebot/commands/developer/ShutdownCommand.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+public class ShutdownCommand implements ICommandRestricted {
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        CascadeBot.logger.info("Shutting down via command!");
+        ShutdownHandler.stop();
+    }
+
+    @Override
+    public String defaultCommand() {
+        return ";shutdown";
```
[the remark this answers is not in the document]
Comment (squeezeboxyt) [reply src/main/java/com/cascadebot/cascadebot/commands/developer/ShutdownCommand.java]: Oopsie

File: src/main/java/com/cascadebot/cascadebot/commands/developer/ShutdownCommand.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+package com.cascadebot.cascadebot.commands.developer;
+
+import com.cascadebot.cascadebot.CascadeBot;
+import com.cascadebot.cascadebot.ShutdownHandler;
+import com.cascadebot.cascadebot.commandmeta.CommandContext;
+import com.cascadebot.cascadebot.commandmeta.CommandType;
+import com.cascadebot.cascadebot.commandmeta.ICommandRestricted;
+import net.dv8tion.jda.core.entities.Member;
+
+public class ShutdownCommand implements ICommandRestricted {
```
[the remark this answers is not in the document]
Comment (squeezeboxyt) [reply src/main/java/com/cascadebot/cascadebot/commands/developer/ShutdownCommand.java]: ok

File: src/main/java/com/cascadebot/cascadebot/commands/core/UserInfoCommand.java [hunk 3]
```diff
@@ -31,18 +32,21 @@ public void onCommand(Member sender, CommandContext context) {
             context.reply("Invalid User!");
             return;
         }
+        if(context.getUser().isBot() == true) {
+            context.reply("User is a bot!");
```
[the remark this answers is not in the document]
Comment (squeezeboxyt) [reply src/main/java/com/cascadebot/cascadebot/commands/core/UserInfoCommand.java]: I'll remove that check then

File: src/main/java/com/cascadebot/cascadebot/commands/core/UserInfoCommand.java [hunk 4]
```diff
@@ -31,18 +32,21 @@ public void onCommand(Member sender, CommandContext context) {
             context.reply("Invalid User!");
             return;
         }
+        if(context.getUser().isBot() == true) {
```
[the remark this answers is not in the document]
Comment (squeezeboxyt) [reply src/main/java/com/cascadebot/cascadebot/commands/core/UserInfoCommand.java]: Good to know for the future thx

File: src/main/java/com/cascadebot/cascadebot/commands/core/UserInfoCommand.java [hunk 5]
```diff
@@ -31,18 +32,21 @@ public void onCommand(Member sender, CommandContext context) {
...
+            context.reply("User is a bot!");
+            return;
+        }
         User user = memberForInfo.getUser();
 
         List<Page> pageList = new ArrayList<>();
         EmbedBuilder builder = new EmbedBuilder();
         builder.setTitle(DiscordUtils.getTag(user));
         builder.setThumbnail(user.getAvatarUrl());
-        builder.addField("Crated Data", "TODO", true);
```
[the remark this answers is not in the document]
Comment (squeezeboxyt) [reply src/main/java/com/cascadebot/cascadebot/commands/core/UserInfoCommand.java]: Will do.

**Comment by binaryoverload** [comment]
We still want a manual shutdown command, the wrapper will just have the ability to force stop also

File: src/main/java/com/cascadebot/cascadebot/commands/core/UserInfoCommand.java [hunk 6]
```diff
@@ -38,11 +39,13 @@ public void onCommand(Member sender, CommandContext context) {
         builder.setTitle(DiscordUtils.getTag(user));
         builder.setThumbnail(user.getAvatarUrl());
         builder.addField("Crated Data", "TODO", true);
-        builder.addField("Join data", "TODO", true);
-
+        builder.addField("Join Data", "TODO", true);
```
Comment (binaryoverload) [inline src/main/java/com/cascadebot/cascadebot/commands/core/UserInfoCommand.java]: When I said add this back, you need to actually fill it out 😂

File: src/main/java/com/cascadebot/cascadebot/commands/developer/ShutdownCommand.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Override
+    public SecurityLevel getCommandLevel() { return SecurityLevel.OWNER; }
+
+    @Override
+    public CommandType getType() {
+        return CommandType.OWNER;
+    }
+
+
```
Comment (binaryoverload) [inline src/main/java/com/cascadebot/cascadebot/commands/developer/ShutdownCommand.java]: Too many lines here, max 1 inbetween

File: src/main/java/com/cascadebot/cascadebot/commands/core/UserInfoCommand.java (hunk 6 above)
Comment (squeezeboxyt) [reply src/main/java/com/cascadebot/cascadebot/commands/core/UserInfoCommand.java]: all done :)

File: src/main/java/com/cascadebot/cascadebot/commandmeta/CommandContext.java [hunk 8]
```diff
@@ -437,7 +437,7 @@ public void sendPagedMessage(List<Page> pages) {
             pageGroup.getPage(pageGroup.getPages()).pageShow(message, pageGroup.getPages(), pageGroup.getPages());
             pageGroup.setCurrentPage(pageGroup.getPages());
         }));
-        channel.sendMessage("\u00A0").queue(sentMessage -> {
+        channel.sendMessage("Paged message loading...").queue(sentMessage -> {
```
Comment (weeryan17) [inline src/main/java/com/cascadebot/cascadebot/commandmeta/CommandContext.java]: Why do we need a message here. it doesn't take like any time at all the update the message with pages?

File: src/main/java/com/cascadebot/cascadebot/commands/core/UserInfoCommand.java [hunk 9]
```diff
@@ -34,15 +41,27 @@ public void onCommand(Member sender, CommandContext context) {
...
+        builder.addField("User Created", user.getCreationTime().format(DateTimeFormatter.RFC_1123_DATE_TIME), true);
+        builder.addField("Join Date", memberForInfo.getJoinDate().format(DateTimeFormatter.RFC_1123_DATE_TIME), true);
+        builder.addField("User ID", user.getId(), true);
+        builder.addField("Status", StringsUtil.toTitleCase(memberForInfo.getOnlineStatus().toString().replace("_", " ")), true);
 
+        Game game = memberForInfo.getGame();
+        if (game != null && !game.isRich()) {
+            String status = "";
+            if (game.isRich()) {
+                // This will require API I think
```
Comment (weeryan17) [inline src/main/java/com/cascadebot/cascadebot/commands/core/UserInfoCommand.java]: make this a todo. i feel like we should support viewing rpc status in userinfo

File: src/main/java/com/cascadebot/cascadebot/commands/developer/ShutdownCommand.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Override
+    public String defaultCommand() {
+        return ";shutdown";
+    }
+
+    @Override
+    public CommandType getType() {
+        return CommandType.DEVELOPER;
```
[the remark this answers is not in the document]
Comment (weeryan17) [reply src/main/java/com/cascadebot/cascadebot/commands/developer/ShutdownCommand.java]: I agree with binary on this one. should be owner

File: src/main/java/com/cascadebot/cascadebot/commands/developer/RestartCommand.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    public String defaultCommand() {
+        return "restart";
+    }
+
+    @Override
+    public SecurityLevel getCommandLevel() { return SecurityLevel.OWNER; }
+
+    @Override
+    public CommandType getType() {
+        return CommandType.DEVELOPER;
```
Comment (weeryan17) [inline src/main/java/com/cascadebot/cascadebot/commands/developer/RestartCommand.java]: Should be owner

File: src/main/java/com/cascadebot/cascadebot/utils/StringsUtil.java [hunk 12]
```diff
@@ -62,4 +62,14 @@ public static String getProgressBar(float percentage, int length, boolean percen
...
                 " " + (percentAfter ? ( Math.round(percentage) + "%") : ""); // If we want we can add a label showing the percentage overall, since this uses physical bars there cannot be a decimal percentage
     }
+
+    /**
+     * Converts a string to lowercase then makes the first character uppercase
+     *
+     * @param string The string to convert to title case
+     * @return The title-cased string
+     */
+    public static String toTitleCase(String string) {
```
Comment (weeryan17) [inline src/main/java/com/cascadebot/cascadebot/utils/StringsUtil.java]: Why do we need a util method for this

File: src/main/java/com/cascadebot/cascadebot/commandmeta/CommandContext.java (hunk 8 above)
Comment (binaryoverload) [reply src/main/java/com/cascadebot/cascadebot/commandmeta/CommandContext.java]: Well we need something, this is just the most sensical thing I could think of

**Pull request merged**